api/apiclient.go

```diff
@@ -326,10 +238,21 @@ func PerferredHost(info *Info) string {
 // if the context is cancelled.
 // TODO(rogpeppe) pass Context into Login (and all API calls) so
 // that this becomes unnecessary.
-func loginWithContext(ctx context.Context, st *state, info *Info) error {
+func loginWithContext(ctx context.Context, loginProvider LoginProvider, st *state) error {
 	result := make(chan error, 1)
 	go func() {
-		result <- st.Login(info.Tag, info.Password, info.Nonce, info.Macaroons)
+		if loginProvider == nil {
+			result <- errors.New("login provider not specified")
+			return
+		}
```

Comment:
This check doesn't need to be in the go routine. It should just be done as part of loginWithContext args checking. 
Would prefer loginProvider to be the 3rd arg.

Reply:
done